projects/sandbox/datagen/datagen/scripts/waveforms.py

```diff
@@ -93,17 +95,32 @@ def main(
     prior, detector_frame_prior = prior()
     params = prior.sample(num_signals)
 
-    signals = generate_gw(
-        params,
-        minimum_frequency,
-        reference_frequency,
-        sample_rate,
-        waveform_duration,
-        waveform_approximant,
-        detector_frame_prior,
-    )
+    if signal_type == "bns":
+        with ProcessPoolExecutor(140) as exe:
```

Comment:
I like the idea of parallelizing this, but I don't think this is doing what you expect: This will submit a single job that will generate all the requested waveforms in one process.

Reply:
So this was a long discussion with Alec on a thread  that incidentally did not include you. I am going to forward that thread to you and hopefully you can see the entire convo.
The crux was that, using concurrent.futures did reduce the waveform generation from days to under 46 min on the hanford box.

Reply:
Ethan's right. This is only helpful if you submit a job for each choice of parameters, submitting one job that generates waveforms for _all_ the parameters won't multiprocess anything, it will just generate all the waveforms in serial in one process.

Reply:
Yes, so I get your point. Not sure why it reduced the waveform generation time so drastically. Or could it be that the hanford box just behaved rather nicely at that particular run. It's kind of puzzling. 
So I will remove the concurrent.futures for now. If we run into bottlenecks in generating BNS in future, we can revisit at that time.